### ISSUE TITLE
Non-joined fonts in N'Ko

### ISSUE THREAD

**Issue by r12a** [w3c/afrlreq#29]
The layout requirements doc describes  the use of non-cursive fonts for book and article titles, etc.  This would correspond to a similar practice in Adlam, for which a Noto Unjoined font exists.

There is no equivalent Noto font for N'Ko, although the usage in Solo Farabado Cisse's book and in http://cormand.huma-num.fr/maninkabiblio/periodiques/silabosoona5.pdf seems to point to this being more than just a legacy font bug.

<img width="734" alt="224297361-2fd35637-3263-428e-bf13-6760b679db3b" src="https://user-images.githubusercontent.com/4839211/224724160-64e98f74-08ee-4874-b875-3159d21cc0f5.png">

Can anyone confirm that this is indeed an intentionally different font/writing style that is in regular use?

**Comment by DD-fwd** [comment]
This style of showing different font style is on all pages of NKO-ENGLISH BOOK by SEBE MARA ߛߍ߬ߓߍ߫ ߡߊ߬ߙߊ. 
![nkoenglishbook7](https://user-images.githubusercontent.com/73085964/224881692-0ae53606-a46f-46ac-9f4a-320bb0b6ab6e.jpg)

**Comment by andjc** [comment]
@r12a There isn't a Nko unjoined Noto font, but I use the same strategy I use for Adlam, rather than using both the joined and unjoined Noto Sans Adlam fonts, I just use the joined version and use css rules to control the joining behaviour.

For my CSS framework supplements for African languages, I have the following class:

```css
.el_unjoined {
    font-feature-settings: "init" 0, "medi" 0, "fina" 0, "isol" 1;
}
```

**Comment by r12a** [comment]
@andjc that's a clever idea, but the shaping is not correct, because many non-joined letters typically  have a small horizontal line at the bottom (see the example in the previous comment).

<img width="149" alt="Screenshot 2023-03-14 at 07 06 27" src="https://user-images.githubusercontent.com/4839211/224922159-cc8b1526-3fe6-4b6f-b0a2-077a666c4fa0.png">

which comes out as 

<img width="144" alt="Screenshot 2023-03-14 at 07 08 56" src="https://user-images.githubusercontent.com/4839211/224922586-3b82a5ca-d910-4da4-8e0d-0df7998fef08.png">

**Comment by donaldsoncd** [comment]
Yes, using unjoined N'ko fonts occurs regularly as a stylistic choice in texts nowadays.

That said, it's my belief that it emerged by in large from connected N'ko fonts not being properly supported across the board and people have incorporated it into their work.

(It doesn't appear the excerpts of the hand-written version Sulemaana Kantè's dictionary that I have, for instance.)

<img width="589" alt="image" src="https://user-images.githubusercontent.com/28364193/224947701-359cfb86-48a3-412a-9ee3-c36a62305b3d.png">


That said, there is some evidence of it being in digitial graphic design as well. For instance in this newpaper below, note the title of the newspaper (but I believe this is drawn -- it's not a font; you can also see the use of the non-connected font to the right for the table of contents):

<img width="860" alt="image" src="https://user-images.githubusercontent.com/28364193/224946217-2d158dda-71cb-44ff-a46d-587650e3d5e7.png">

So, perhaps, it can be considered a graphic design choice now, but I believe it's done via work-arounds with legacy fonts that aren't properly rendered.

**Comment by andjc** [comment]
@r12a an interesting issue, what your image seems to indicate is that the typeface used is using a mixture of isolated and initial forms of glyphs, if I go by the glyph  repertoire  in Noto Sans Nko.

So a few questions:

1) Is Noto's implementation of isolated glyphs flawed?
2) Are the glyphs in the unjoined form different from the isolated glyphs in the  joined form, and if yes, what would the normative glyph be for each character. What is the relationship between unjoined glyphs and isolated glyphs? Are we dealing with fundamental differences in the design of glyphs?
3) Are the differences a product of the typeface used in publication (ie is it a non-normative mix of isolated and initial forms?

If it's 2) then we may need to rethink what an isolated glyph really is.  If 2) is the correct scenario then we need isolated, initial, medial, final and unjoined forms (i.e. there are 5 potential glyphs), and unjoined and isolated are distinct (different).

or 

4) the typeface reflects a design choice that is non-normative, but acceptable (ie isolated and initial are sometimes interchangeable).

**Comment by andjc** [comment]
Looking at images provided by @r12a and @DD-fwd the unjoined headings seem to be a mix of isolated and initial glyphs, while the newspaper image provided by @donaldsoncd would appear to be isolated glyphs in a slab serif style design.

I am wondering if the unjoined text should be seen more as a distinction between sans-serif and serif?

**Comment by r12a** [comment]
If we agree that the unjoined style has become a regular stylistic choice, i'll raise an issue in the Noto repository to suggest that they consider making a font, like they did for Adlam.  There is no issue at the moment that addresses that (open or closed). See https://notofonts.github.io/#nko

**Comment by andjc** [comment]
There seems to be enough examples of its use, but between the examples in this issue, there seem to be two approaches:

1. a mix of isolated and initial glyphs (and the question for this is which character gets which style of glyph)
2. a serif version of the isolated glyphs

Honestly, I don't think we have enough information to raise an issue. If we recommend an unjoined font, what glyphs should it use, the developers will need guidance. And how does an unjoined sans serif font differ from a joined font with joining behaviour disabled? Are the differences we see real distinctions,  accidental, or a choice of the designer/caligrapher? Is there a unified approach or do differnet designers/caligraphers vary?

And should the joined font be also able to support the unjoined version?

**Comment by NeilSureshPatel** [comment]
I think the way unconnected forms are formed is a bit muddy. The legacy fonts often have the baseline stroke on the forms because of how the fonts were built without OpenType features. The baseline stroke gives some appearance of a connected script. As a result this has become a kind of style. When we worked with WODN to develop Kigelia we made the isolated forms without the baseline stroke, which was what they approved. This is in line with other examples of isolated forms being used. For example in the header of kanjamadi.com.

<img width="674" alt="image" src="https://user-images.githubusercontent.com/46502494/225019106-98f62a14-626f-4421-8058-397971706b45.png">

This is also seen in Dalby's catalogue of characters.

<img width="455" alt="image" src="https://user-images.githubusercontent.com/46502494/225019990-cf5e2d40-fbf5-486b-ba0a-c0b98439660d.png">

I have recently received all of the legacy fonts developed by WODN. These are Arabic-encoded fonts. A quick look into a few of these I can see that the isolated forms also don't have the baseline stroke included.

<img width="732" alt="image" src="https://user-images.githubusercontent.com/46502494/225025917-b5dd2424-2f92-4f54-ad6c-7f376055caa3.png">


The other thing to keep in mind is that setting type on curve for headers is also very common. Since joining doesn't work very well in these cases the isolated style with a baseline stroke bridges the gap. I think the isolated forms with a baseline stroke has just evolved to become a stylistic option due to lack of proper fonts and previous lack of support.

On a related note, I feel that we shouldn't have both unconnected and connected fonts but use the isol feature as a stylistic switch much like we do for Bold and Italic. I haven't gotten very far with that discussion though. Since there are not any standalone unconnected fonts or an easy way to switch on the isol feature the old fonts with the baseline stroke on isolated forms have become normalized to some extent.

**Comment by r12a** [comment]
fwiw (probably not much) i made a little table that compares isolated forms from Noto and from some unjoined fonts i found on the Web.  See https://r12a.github.io/scripts/nkoo/nqo.html#noncursive and click on the line with the black triangle.

**Comment by NeilSureshPatel** [comment]
@r12a I did receive most of the Latin-encoded legacy fonts not long ago. We have been asked to clean them up with proper encoding and shaping. I haven't reviewed all of them yet, but I can send you a couple if you want, so you can fill in the missing slots and get clearer images. Let me know.

**Comment by r12a** [comment]
@NeilSureshPatel yes please. Thanks.

**Comment by r12a** [comment]
I added a brief description of unjoined glyphs to https://w3c.github.io/afrlreq/nko/index.html#noncursive

**Comment by andjc** [comment]
@r12a it may be worth determining if different fonts (from different) designers add the bar to the same set of glyphs, i.e. how consistent are the sets of glyphs between fonts?

**Comment by NeilSureshPatel** [comment]
I think perhaps we are putting too much weight on the presence of the underline on some glyphs in the Latin-encoded legacy fonts. Every one of those that are in use were designed by Dr. Mamadi Diane who is also a key member of the World Organization for the Development of N'ko (WODN). WODN and Dr. Diane have been pretty clear that those are old designs without any shaping behavior and that the connecting stroke should not be present on isolated forms. Any variation in how Dr. Diane applied the under stroke is a workaround he devised for a certain application. We have been asked to make all of Diane's fonts Unicode compliant and fix those legacy implementation issues. 

There are only three Unicode compliant fonts in use, Noto Sans N'ko, Ebrima, and Kigelia all of which don't use a connecting stroke on isolated forms. I don't know the history of Noto Sans N'ko but the other two fonts were designed with input from WODN. 

I guess my point is that there really isn't a lot of people making N'ko fonts. I think it makes sense to document that connecting strokes may be seen isolated forms, but they aren't considered to be canonical. It is kind of like swashes on Latin letters; yes, they exist in display applications, but we don't imply that there is standard to how they appear.

**Comment by andjc** [comment]
@NeilSureshPatel  that aligns with my impression and why in the CSS I use I force the isolated forms of the three fonts.

**Comment by donaldsoncd** [comment]
I share @NeilSureshPatel's assessment from my own work and research with Diané and other N'ko activists/authors.
